lib/restbase.js

```diff
@@ -67,10 +76,25 @@ function RESTBase(options, req) {
         this.rb_config = options.conf;
         this.rb_config.user_agent = this.rb_config.user_agent || 'RESTBase';
         this._rootReq = null;
+        this._forwardedCookies = null;
+        this._accessRestrictions = {};
     }
-
 }
 
+RESTBase.prototype.isInternalRequest = function(req) {
+    if (this.rb_config.internal_request_whitelist) {
+        var uri = req.uri.toString();
+        return this.rb_config.internal_request_whitelist.some(function(regex) {
+            if (/^\/.+\/$/.test(regex)) {
+                return new RegExp(regex.substring(1, regex.length - 1)).test(uri);
```

Comment:
It would be cheaper to build a single regexp once at startup, and then just call that.

Example: 

```javascript
var regexps = ['/foo/', '/bar/'].map(function(s) {
  new RegExp(return s.substring(1, regex.length - 1)).toString();
});
var whitelist = new RegExp(regexps.join('|'));
```